api/v1alpha1/common_types.go

```diff
@@ -0,0 +1,544 @@
+)
+
+const (
+	// Finalizer is the finalizer string for resoures in the saas group
+	Finalizer string = "saas.3scale.net"
+)
+
+// ImageSpec defines the image for the component
+type ImageSpec struct {
+	// Docker repository of the image
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Name *string `json:"name,omitempty"`
+	// Image tag
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Tag *string `json:"tag,omitempty"`
+	// Name of the Secret that holds quay.io credentials to access
+	// the image repository
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	PullSecretName *string `json:"pullSecretName,omitempty"`
+}
+
+type defaultImageSpec struct {
+	Name, Tag, PullSecretName *string
+}
+
+// Default sets default values for any value not specifically set in the ImageSpec struct
+func (spec *ImageSpec) Default(def defaultImageSpec) {
+	spec.Name = stringOrDefault(spec.Name, def.Name)
+	spec.Tag = stringOrDefault(spec.Tag, def.Tag)
+	spec.PullSecretName = stringOrDefault(spec.PullSecretName, def.PullSecretName)
+}
+
+// IsDeactivated true if the field is set with the deactivated value (empty struct)
+func (spec *ImageSpec) IsDeactivated() bool { return false }
+
+// InitializeImageSpec initializes a HTTPProbeSpec struct
+func InitializeImageSpec(spec *ImageSpec, def defaultImageSpec) *ImageSpec {
+	if spec == nil {
+		new := &ImageSpec{}
+		new.Default(def)
+		return new
+	}
+	copy := spec.DeepCopy()
+	copy.Default(def)
+	return copy
+}
+
+// HTTPProbeSpec specifies configuration for an HTTP probe
+type HTTPProbeSpec struct {
+	// Number of seconds after the container has started before liveness probes are initiated
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	InitialDelaySeconds *int32 `json:"initialDelaySeconds,omitempty"`
+	// Number of seconds after which the probe times out
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	TimeoutSeconds *int32 `json:"timeoutSeconds,omitempty"`
+	// How often (in seconds) to perform the probe
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	PeriodSeconds *int32 `json:"periodSeconds,omitempty"`
+	// Minimum consecutive successes for the probe to be considered successful after having failed
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	SuccessThreshold *int32 `json:"successThreshold,omitempty"`
+	// Minimum consecutive failures for the probe to be considered failed after having succeeded
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	FailureThreshold *int32 `json:"failureThreshold,omitempty"`
+}
+
+type defaultHTTPProbeSpec struct {
+	InitialDelaySeconds, TimeoutSeconds, PeriodSeconds,
+	SuccessThreshold, FailureThreshold *int32
+}
+
+// Default sets default values for any value not specifically set in the HTTPProbeSpec struct
+func (spec *HTTPProbeSpec) Default(def defaultHTTPProbeSpec) {
```

Comment:
Should be this function called (and related functions), called `ProbeSpec` instead of `HTTPProbeSpec`?

It is being used for both http/tcp liveness/readiness because its fields are common (InitialDelaySeconds, TimeoutSeconds, PeriodSeconds, SuccessThreshold, FailureThreshold)

Reply:
Uhm sure, I'll rename it.

Reply:
Renamed in 0a08eb400551a1b073c22e5615ecbc49941e8768

Reply:
After the rename, there are still plenty of `defaultHTTPProbeSpec` where it should be changed to `defaultProbeSpec`, isn't it?

For the ones already changed, remember to change the comment also (a part from funstion name), like for example (there are a few):
```
// ProbeSpec specifies configuration for an HTTP probe
```

Should be something like:
```
// ProbeSpec specifies configuration for an Probe
```

Reply:
@slopezz I think all is renamed now

Reply:
Looks nice!